test/src/tests-session-manager.ts

```diff
@@ -0,0 +1,607 @@
+        });
+
+        it('should update dateLastEventSent if sessionStartDate does not exist', () => {
+            mParticle.init(apiKey, window.mParticle.config);
+
+            const mpInstance = mParticle.getInstance();
+
+            mpInstance._Store.sessionStartDate = null;
+            mpInstance._SessionManager.startNewSession();
+
+            expect(mpInstance._Store.dateLastEventSent).to.eql(now);
+        });
+
+        it('should NOT update dateLastEventSent if sessionStartDate already exists', () => {
+            const dateInThePast = new Date(2015, 10, 21);
+
+            mParticle.init(apiKey, window.mParticle.config);
+
+            const mpInstance = mParticle.getInstance();
+
+            mpInstance._Store.sessionStartDate = dateInThePast;
+            mpInstance._SessionManager.startNewSession();
+
+            expect(mpInstance._Store.dateLastEventSent).to.eql(now);
+        });
+
+        it('should call Identity.identify if Store.identifyCalled is false', () => {
+            // Init already an identify request and resets the identiy callback,
+            // To test this feature in isolation we will "reset" the identify
+            // request and callback after init so we can test startNewSession
+            // in isolation.
+
+            const identityApiData: IdentityApiData = {
+                userIdentities: {
+                    customerid: 'my-customer-id',
+                },
+            };
+
+            mParticle.init(apiKey, window.mParticle.config);
+
+            const mpInstance = mParticle.getInstance();
+            const callbackSpy = sinon.spy();
+            const identifySpy = sinon.spy(mpInstance.Identity, 'identify');
+
+            mpInstance._Store.SDKConfig.identifyRequest = identityApiData;
+            mpInstance._Store.SDKConfig.identityCallback = callbackSpy;
+            mpInstance._Store.identifyCalled = false;
+            //
+            mpInstance._SessionManager.startNewSession();
+
+            expect(identifySpy.getCall(0).args[0]).to.eql(identityApiData);
+            expect(identifySpy.getCall(0).args[1]).to.eql(callbackSpy);
+            expect(callbackSpy.called).to.equal(true);
+            expect(mpInstance._Store.identifyCalled).to.eql(true);
+            expect(mpInstance._Store.SDKConfig.identityCallback).to.eql(null);
+        });
+    });
+
+    describe('#endSession', () => {
+        // TODO: Set up scaffolding for session
+        it.skip('should end a session', () => {
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+            const persistenceSpy = sinon.spy(mpInstance._Persistence, 'update');
+
+            mpInstance._SessionManager.endSession();
+
+            expect(mpInstance._Store.sessionId).to.equal(null);
+            expect(mpInstance._Store.dateLastEventSent).to.equal(null);
+            expect(mpInstance._Store.sessionAttributes).to.eql({});
```

Comment:
I think to be more clear, these should be shown to have actual values prior to line 254.

Reply:
To be handled as part of https://go.mparticle.com/work/SQDSDKS-5614